scripts/globals/garrison_data.lua

```diff
@@ -0,0 +1,673 @@
+-----------------------------------
+-- Garrison Data
+-----------------------------------
+require("scripts/globals/zone")
+-----------------------------------
+xi = xi or {}
+xi.garrison = xi.garrison or {}
+
+-- Name is Determined by Nation and LevelCap
+-- Names in order of xi.nation values (sandoria, bastok, windurst)
+xi.garrison.allyNames =
+{
+    [20] = { "Patrician", "Recruit", "Candidate" },
+    [30] = { "Trader", "Mariner", "Scholar" },
+    -- TODO: Capture Windy npc. Shouldn't be TempleKnight.
+    [40] = { "TempleKnight", "GoldMusketeer", "TempleKnight" },
+    [50] = { "RoyalGuard", "GoldMusketeer", "Patriarch" },
+    [75] = { "MilitaryAttache", "MilitaryAttache", "MilitaryAttache" },
+}
+
+-- Group Ids are different per cap due to min / max level requirements
+-- They all use the same pool at the moment, but we could also change families
+-- based on cap, which would change base stats
+xi.garrison.allyGroupIds =
+{
+    [20] = 1,
+    [30] = 2,
+    [40] = 3,
+    [50] = 4,
+    [75] = 5,
+}
+
+-- This is used to replace blank weapons in ally looks.
+-- TODO: Use proper npc looks for all models or at least find the right range
+-- so we can purely generate these
+xi.garrison.allyArsenal =
+{
+    "08",
+    "83",
+    "B8",
+    "73",
+    "62",
+    "01",
+    "19",
+    "6B",
+    "B5",
+    "75",
+    "6F",
+    "BB",
+    "3C",
+    "56",
+    "4E",
+    "19",
+    "5C",
+    "61",
+    "E6"
+}
+
+-- Look is Determined by Nation and LevelCap (Appears to be 4 for each outpost need more data though)
+xi.garrison.allyLooks =
+{
+    -- Level 20 garrison looks
+    -- The looks that have not been obtained from captures are commented
```

Comment:
👀

Reply:
I will do what I can to obtain the missing data, but there is a lot. Is there any room for  compromise, such as blocking the areas that are missing captures, so at least some of the areas are usable? If not, I think requiring this to be complete before accepting any enhancements is an impossibly high bar and it will probably never be done.

This is one of the earliest pieces of content that still isn't properly supported. I think finishing Garrison would go a long way to completing emulation for the base game. I'm also willing to refactor what's here into a form that you prefer.

Reply:
> as blocking the areas that are missing captures
Yeah, this will be fine.

> An impossibly high bar
Can I remind you that you've opened a PR called `Implements Garrison`, using 10 commits of previous contributors' work? If you don't want to scope the whole feature, change the title and just submit a single confirmed-working Garrison at a time - alongside any system and core changes needed as you go.

The reason I'm so aggressive about this is incorrect work gets fixed/finished _even less_ than partially completed work. If it's a feature that not many people care about and it's a vague approximation of what it's supposed to be - then it'll be left in that state forever. While you're here with the code fresh in your mind and an understanding of the feature - that's the perfect time to actually implement it properly.

Also, upstream holds itself to higher quality and correctness standards than downstream - because everyone builds off of us. More servers and groups than just ASB and Horizon use it, and it's unfair to them to just yeet half-finished work into the repo.

Reply:
Thanks for your feedback, which I genuinely appreciate. I was not expecting to push things in the original state, and your comments allowed me to more easily identify the areas for improvement.

I hope I've alleviated some concerns by properly documenting the features and moving custom behavior to a module. Regarding code, I agree on high standards of correctness and quality without hesitation, the bar referred only to the completeness of captures, which are challenging to obtain for this content (One of the reasons, it's still so bare).

With this enhancements to this system, I'm optimistic the missing capture data will begin to be submitted, as it will unlock those areas for immediate use on live servers.

If there are any further concerns about the implementation, I will gladly look into resolving them.

Reply:
I'm optimistic that models aren't difficult to acquire from people with retail access before hand.

I wish I had a cap suite that worked on ashita instead of widower, I'd go suicide on garrison for you all week long For now I've made a request and on [captures discord](https://discord.com/channels/443544205206355968/443893540922064896/1133447193441865778), where I see you are already a member.

Also this is a thing, but almost nothing is properly labelled: https://github.com/LandSandBoat/lsb-wiki/blob/main/Model-ID-Reference.md

I've made [an appeal](https://discord.com/channels/392903136336936960/883227930828886060/1133184440688984085) to the wider community hoping to find people interested in fixing that up, but I am less optimistic about that than I am captures.

> he bar referred only to the completeness of captures, which are challenging to obtain for this content (One of the reasons, it's still so bare).

You kinda set that bar yourself at 1st (per Zach's reply, doesn't all have to happen at once), though I do not believe that was "impossibly" high. If you want to do parts label your pr and commits for your parts if it doesn't "work" without missing parts that's ok.\* If need be messages can be edited while still retaining author info.

<sub><sup>\*</sup> if something "sorta works" even when its wrong ppl are less encouraged to fill it in correctly _and_ the proj gets blamed for it being wrong by the ppl that actually pay attention to that. I could write and entire angry editorial about how many times I've seen the teams (plural, multiple iterations since dsp but to some people we're "all the same") get disparaging comments for it which resulted in the current crafting table and its missing crafts after a previous project lead had enough of that. </sub>

Reply:
> models aren't difficult to acquire from people with retail access before hand

While I respect the sentiment, here are the reasons why Garrison captures are especially challenging:
* Players can only participate in Garrison once per week
* The player's nation *must* control the current region. This may be practically impossible for some region/nation combos depending on the server.
* The number of NPCs spawned is based on the number of *real* players (Guaranteeing a complete capture would mean getting 18 players to do this obsolete content, good luck with that)
* There are 15 different nation/level/region combinations that each need to meet these very tough requirements, that can only be captured one week at a time.

Outside of Chocobo raising, this is probably one of the worst pieces of content capture wise. And at least you can solo Chocobo raising (Though appreciate there's more content *to* capture with raising). To get the most productive captures, you can't solo Garrison.

Reply:
gosh I has no idea! I am SO shocked!

-_-

Do you really want me to explain to you why retorting like this is a bad idea?

Reply:
> The number of NPCs spawned is based on the number of real players (Guaranteeing a complete capture would mean getting 18 players to do this obsolete content, good luck with that)

This one we haven't proven yet (#shitwikisays, etc.). We think it's a function of both total players (Trusts can't be called, unsure about Adventuring fellows) and the number of parties involved. Is a 6-player party going to spawn as many waves of certain sizes as 3x parties of 2? etc.

Reply:
That's a good point. If I can get a few people to help test as an alliance (1 per party), I will make the adjustments to that function if required.

Reply:
> > The number of NPCs spawned is based on the number of real players (Guaranteeing a complete capture would mean getting 18 players to do this obsolete content, good luck with that)
> 
> This one we haven't proven yet (#shitwikisays, etc.). We think it's a function of both total players (Trusts can't be called, unsure about Adventuring fellows) and the number of parties involved. Is a 6-player party going to spawn as many waves of certain sizes as 3x parties of 2? etc.

I know it _used to_ just count parties (I had tested it about 2011 or so) and also SE rarely updates older content without some announcement of it (like stripping level caps) so its a good possibility its  still working that way. 

Alas, if my system had good capture tools (ashita..) and I had enough days off it I could also nation change and cap regions one per week as needed (I can drum up 17 warm bodies easy enough) as long as the win event data isn't needed on a max size alliance (17 winners I can _not_ drum up lol)

Reply:
worth noting more modern advanced jobs (blu, pup, run) likely go a long way on these as the content was designed before they existed and wouldn't have been balanced with them in mind.